p8e-api/src/main/kotlin/io/provenance/engine/service/ChaincodeInvokeService.kt

```diff
@@ -49,7 +48,37 @@ class ChaincodeInvokeService(
     private val chaincodeProperties: ChaincodeProperties,
     private val provenanceGrpc: ProvenanceGrpcService,
 ) : IChaincodeInvokeService {
-    private val log = logger()
+
+    companion object {
+        private val log = logger()
+
+        private val blockScopeIds = HashSet<String>()
+        private val scopeLockHeights = HashMap<String, Long>()
```

Comment:
We should probably move these to ConcurrentHashMaps now. Previously only the batching thread was using it, but now it's being unlocked in the event stream thread and reaper as well.

Reply:
oh good point, I'll update that

Reply:
updated